includes/class-wc-stripe-intent-controller.php

```diff
@@ -956,7 +956,7 @@ public function is_mandate_data_required( $selected_payment_type, $is_using_save
 	public function create_and_confirm_setup_intent( $payment_information ) {
 		$request = [
 			'payment_method'       => $payment_information['payment_method'],
-			'payment_method_types' => $payment_information['payment_method_types'],
+			'payment_method_types' => $payment_information['payment_method_types'] ?? [ $payment_information['selected_payment_type'] ],
```

Comment:
@wjrosa this fixes the `"message": "The type of the provided PaymentMethod (pm_1PgpCJGwEEw4tjx00ZY6OPTb) is link. This does not match the expected type card. Try confirming with a PaymentMethod of type card.",` error you mentioned to me in Slack. 

In https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3310 I changed this to use the provided `payment_method_types`, however this add payment method flow didn't provide that arg. see [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/fix/3224-link-payments-require-mandate/includes/class-wc-stripe-intent-controller.php#L1010-L1016).